pipeline_tools/input_utils.py

```diff
@@ -1,219 +1,181 @@
+         tuple: A tuple consisting of the url to the input fastq files, which are corresponding to read index 1 and 2
+         respectively.
     """
-    if version.startswith('4.'):
-        return _get_input_metadata_file_uuid_v4(manifest_files)
-    return _get_input_metadata_file_uuid_v5_or_higher(manifest_files)
+    fastq_1_url = fastq_2_url = None
+    sequence_files = bundle.sequencing_output
 
+    for sf in sequence_files:
+        if fastq_1_url and fastq_2_url:
+            return fastq_1_url, fastq_2_url  # early termination
+        if sf.read_index == 'read1':
+            fastq_1_url = sf.manifest_entry.url
+        if sf.read_index == 'read2':
+            fastq_2_url = sf.manifest_entry.url
+    return fastq_1_url, fastq_2_url
 
-def _get_input_metadata_file_uuid_v5_or_higher(manifest_files):
-    """Get the uuid of the files.json file"""
-    return dcp_utils.get_file_uuid(manifest_files, 'file.json')
 
+def download_file(item, dss_url, http_requests=HttpRequests()):
+    """Download the metadata for a given bundle from the HCA data store (DSS).
 
-def _get_input_metadata_file_uuid_v4(manifest_files):
-    """Get the uuid of the assay.json file"""
-    return dcp_utils.get_file_uuid(manifest_files, 'assay.json')
-
-
-def get_sample_id_file_uuid(manifest_files, version):
-    """Get the uuid of the file containing the sample id,
-    e.g. assay.json in v4
+    This function borrows a lot of existing code from the `metadata-api` code base for consistency,
+    and this won't be required after migrating to use the HCA DSS Python API `dss_client` directly.
 
     Args:
-        manifest_files (dict): file metadata
-        version (str): version of metadata
-
-    Returns:
-        String giving uuid of file containing sample id
-
-    Raises:
-        NotImplementedError: if metadata version is unsupported
+        item (typing.ItemsView): A dictionary's ItemsView object consisting of file_name and the manifest_entry.
+        dss_url (str): The url for the DCP Data Storage Service.
+        http_requests (HttpRequests): The HttpRequests object to use.
     """
-    if version.startswith('4.'):
-        return _get_sample_id_file_uuid_v4(manifest_files)
-    else:
-        return _get_sample_id_file_uuid_v5_or_higher(manifest_files)
-        # raise NotImplementedError('Only implemented for v4 and v5 metadata')
-
-
-def _get_sample_id_file_uuid_v5_or_higher(manifest_files):
-    """Get the uuid of the links.json file"""
-    return dcp_utils.get_file_uuid(manifest_files, 'links.json')
-
+    file_name, manifest_entry = item
+    file_uuid = manifest_entry['uuid']
+    return file_name, dcp_utils.get_file_by_uuid(file_id=file_uuid, dss_url=dss_url, http_requests=http_requests)
 
-def _get_sample_id_file_uuid_v4(manifest_files):
-    """Get the uuid of the assay.json file"""
-    return dcp_utils.get_file_uuid(manifest_files, 'assay.json')
 
+def get_metadata_files(metadata_files_dict, dss_url, num_workers=None):
```

Comment:
Rather than downloading the metadata as individual files, could we just make one call to query blue box to return all metadata as one big json blob?

Reply:
Based on an offline discussion with @dshiga The current function is copied over from the [`metadata-api`](https://github.com/HumanCellAtlas/metadata-api/blob/1b7192cecbef43b5befecc4153bf2e2f4db5bb16/src/humancellatlas/data/metadata/helpers/dss.py#L14), I will make a ticket for this as a future work. The comments here totally make sense, otherwise we can potentially run into some capabilities problems in the future. Also, `metadata-api` repo might be a better palce to handle this logic, once we start to use the `dss_client`